### PR TITLE
RedisStore: Update init and Documentation updates

### PR DESCRIPTION
**Pull request by eyurtsev**
* Update Redis Store to support init from parameters
* Update notebook to show how to use redis store, and some fixes in documentation

### CODE REVIEW

File: libs/langchain/langchain/storage/redis.py [hunk 1]
```diff
@@ -1,7 +1,10 @@
-from typing import Any, Iterator, List, Optional, Sequence, Tuple, cast
+from typing import TYPE_CHECKING, Iterator, List, Optional, Sequence, Tuple, cast
 
 from langchain.schema import BaseStore
 
+if TYPE_CHECKING:
```
Comment (eyurtsev) [inline libs/langchain/langchain/storage/redis.py:5]: @baskaryan what's our approach for type hinting optional code that is on the unit test path?

Comment (baskaryan) [reply libs/langchain/langchain/storage/redis.py:5]: does this cause issues? wouldn't it skip this during unit test

Comment (baskaryan) [reply libs/langchain/langchain/storage/redis.py:5]: think you just need a 
```python
from __future__ import annotations
```
 in there

Comment (eyurtsev) [reply libs/langchain/langchain/storage/redis.py]: Added other stuff to this PR and removed the type check for now -- can follow up in a bit

**Review by baskaryan — APPROVED** [review]
ship it

**Pull request merged**